### DISCUSSION TITLE
Performance consulting

### DISCUSSION THREAD

**Discussion by aftersnow** [vectordotdev/vector#15495] [Q&A] [answered]
We are interested in Vector, considering using Vector in out production environments. We saw the performance tests posted here:
https://github.com/vectordotdev/vector-test-harness/tree/master/cases/file_to_tcp_performance
The Vector's File to TCP  disk thrpt is 76.7MiB/s.
Is the test running in the VM environments?  And what is the number of this VM CPU cores and memory? (I didn't found the info)

**Comment by tobz** [comment, marked as answer]
Hi @aftersnow!

The benchmarks in `vector-test-harness` are very old at this point, and likely out-of-date, in terms of both Vector's performance as well as competitors. I would not use those numbers as a reference.

We're working on some better benchmarks, similar to `vector-test-harness`, that we can expose in an ongoing fashion i.e. run them every night on `master`, per release, etc. This isn't ready quite yet, though, so unfortunately, I think the best course of action would be to work towards load testing Vector within your own environment.

We're certainly happy to try and answer questions around unexpected performance issues, or things that seem like they are performance issues, but we can't confident (yet, anyways) tell you that Vector does worse/the same/better than a competitor on a specific hardware configuration, or exactly how many bytes/second you be able to expect on a specific hardware configuration.

**Reply by aftersnow** [reply]
OK, thanks very much for your answer!
